autogpt_platform/backend/backend/util/file.py

```diff
@@ -0,0 +1,141 @@
+             We realpath-check so no symlink or '..' can escape the folder.
+
+    :param exec_id:        Unique identifier for the execution context.
+    :param file:           Data URI, URL, or local (relative) path.
+    :param return_content: If True, return a data URI of the file content.
+                           If False, return the *relative* path inside the exec_id folder.
+    :return:               The requested result: data URI or relative path.
+    """
+
+    # 1) Build the absolute base path for this exec_id
+    temp_base = get_path(exec_id, "")
+    os.makedirs(temp_base, exist_ok=True)
+
+    # 2) Helper functions
+    def _extension_from_mime(mime: str) -> str:
+        ext = mimetypes.guess_extension(mime, strict=False)
+        return ext if ext else ".bin"
+
+    def _file_to_data_uri(path: str) -> str:
+        mime_type, _ = mimetypes.guess_type(path)
+        if not mime_type:
+            mime_type = "application/octet-stream"
+        with open(path, "rb") as f:
+            raw = f.read()
+        b64 = base64.b64encode(raw).decode("utf-8")
+        return f"data:{mime_type};base64,{b64}"
+
+    def _strip_base_prefix(absolute_path: str) -> str:
+        # Stripe temp_base prefix and normalize path
+        return absolute_path.removeprefix(temp_base).removeprefix(os.sep)
+
+    def _ensure_inside_base(path_candidate: str) -> str:
+        """
+        Resolve symlinks via realpath and ensure the result is still under temp_base.
+        If valid, returns the real, absolute path.
+        Otherwise, raises ValueError.
+        """
+        real_candidate = os.path.realpath(path_candidate)
+        real_base = os.path.realpath(temp_base)
+        # Must be either exactly the folder or inside it
+        if (
+            not real_candidate.startswith(real_base + os.sep)
+            and real_candidate != real_base
+        ):
+            raise ValueError(
+                "Local file path is outside the temp_base directory. Access denied."
+            )
+        return real_candidate
+
+    # 3) Distinguish between data URI, URL, or local path
+    if file.startswith("data:"):
+        # === Data URI ===
+        match = re.match(r"^data:([^;]+);base64,(.*)$", file, re.DOTALL)
+        if not match:
+            raise ValueError(
+                "Invalid data URI format. Expected data:<mime>;base64,<data>"
+            )
+
+        mime_type = match.group(1).strip().lower()
+        b64_content = match.group(2).strip()
+
+        # Generate random filename with guessed extension
+        extension = _extension_from_mime(mime_type)
+        local_filename = str(uuid.uuid4()) + extension
+        # Our intended path
+        intended_path = os.path.join(temp_base, local_filename)
+        absolute_path = _ensure_inside_base(intended_path)
+
+        # Decode and write
+        raw_bytes = base64.b64decode(b64_content)
+        with open(absolute_path, "wb") as f:
+            f.write(raw_bytes)
```

Comment:
when are these files cleaned up?

Reply:
Periodic cleanup by the OS if any.
I've added explicit removal by the end of the execution.